src/Sentry/Infrastructure/TraceDiagnosticLogger.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Diagnostics;
+using Sentry.Extensibility;
+using System.Linq;
+
+namespace Sentry.Infrastructure
+{
+    /// <summary>
+    /// Trace logger used by the SDK to report its internal logging.
+    /// </summary>
+    /// <remarks>
+    /// Logger available when hooked to an IDE. It's useful when debugging apps running under IIS which have no output to Console logger.
+    /// </remarks>
+    public class TraceDiagnosticLogger : IDiagnosticLogger
+    {
+        private readonly SentryLevel _minimalLevel;
+
+        private Lazy<TraceListener?> _traceListener = new Lazy<TraceListener?>(() => GetFirstOrDefaultListener());
+
+        /// <summary>
+        /// Get the first TraceListener, usually it's the process debug output.
+        /// </summary>
+        /// <returns>The first TraceListener, if available</returns>
+        private static TraceListener? GetFirstOrDefaultListener()
+        {
+            var listeners = Trace.Listeners;
+            return listeners.Count > 0 ? listeners[0] : null;
```

Comment:
It works, not a fan of this line and linq doesn't work with listeners.

Reply:
Maybe we should write to **all** listeners, not just one?